parsec/parsec_mpi_funnelled.c

```diff
@@ -109,33 +109,23 @@ static inline int next_tag(int k) {
 #define MAX_DYNAMIC_REQ_RANGE 30 /* according to current implementation */
 #define EACH_STATIC_REQ_RANGE 5 /* for each registered tag */
 
-/* Hash table for tag_structure. Each registered tags will be book-kept
- * using this structure.
- */
-static int tag_hash_table_size = 1<<MPI_FUNNELLED_MAX_TAG; /**< Default tag hash table size */
-static parsec_hash_table_t *tag_hash_table = NULL;
-
-static parsec_key_fn_t tag_key_fns = {
-    .key_equal = parsec_hash_table_generic_64bits_key_equal,
-    .key_print = parsec_hash_table_generic_64bits_key_print,
-    .key_hash  = parsec_hash_table_generic_64bits_key_hash
-};
-
 typedef struct mpi_funnelled_tag_s {
-    parsec_hash_table_item_t  ht_item;
     parsec_ce_tag_t tag; /* tag user wants to register */
-    char **buf; /* Buffer where we will receive msg for each TAG
-                 * there will be EACH_STATIC_REQ_RANGE buffers
-                 * each of size msg_length.
-                 */
     int start_idx; /* Records the starting index for every TAG
                     * to unregister from the array_of_[requests/indices/statuses]
                     */
     size_t msg_length; /* Maximum length allowed to send for each
                         * registered TAG.
                         */
+    char **buf; /* Buffer where we will receive msg for each TAG
+                 * there will be EACH_STATIC_REQ_RANGE buffers
+                 * each of size msg_length.
+                 */
 } mpi_funnelled_tag_t;
 
+#define PARSEC_MAX_REGISTERED_TAGS  16
```

Comment:
16 seems a bit low if we expose this to the outside world. Why not 1k?

Reply:
16 is at least 3 times more than PaRSEC needs right now. We can scale it up once the need arise.